### PR TITLE
add salt:// support to salt-cloud for .map files

### PR DESCRIPTION
**Pull request by SaltyCharles**
add salt:// support to salt-cloud for .map files

### CODE REVIEW

File: salt/cloud/__init__.py [hunk 1]
```diff
@@ -1670,17 +1670,26 @@ def read(self):
         if self.opts.get('map', None) is None:
             return {}
 
+        local_minion_opts = self.opts.copy()
```
Comment (cachedout) [inline salt/cloud/__init__.py]: You probably want a `copy.deepcopy()` here.

**Comment by cachedout** [comment]
@SaltyCharles We'll need this rebased before we can merge it and I also left one comment in-line.

@techhat Could you also take a look here please?

**Comment by cachedout** [comment]
@techhat I would still like you to take a look here if you could. Thanks!

**Comment by dyaln** [comment]
When testing, I ran across a failure from 827e09f

> AttributeError: 'builtin_function_or_method' object has no attribute 'deepcopy'

This seems to be solved with a change to 
`        local_minion_opts = copy.deepcopy(self.opts)`

**Comment by jfindlay** [comment]
@dyaln, fixed in https://github.com/saltstack/salt/pull/33736.

**Pull request merged**